### PR TITLE
Refactored NotifyBlocker() (and several overloads) into ActorExts

### PR DESCRIPTION
**Pull request by hktk1f40**
Moved the notification of blocking actors (INotifyBlockingMove) into Actor class, it was used in three different places more or less the same way.

### CODE REVIEW

File: OpenRA.Game/Actor.cs [hunk 1]
```diff
@@ -249,6 +249,25 @@ public void Kill(Actor attacker)
 			health.Value.InflictDamage(this, attacker, health.Value.MaxHP, null, true);
 		}
 
+		public void NotifyBlocker(IEnumerable<Actor> units)
```
Comment (Mailaender) [inline OpenRA.Game/Actor.cs]: Maybe call this blockers to make it clear.

**Comment by Mailaender** [comment]
Looking good :+1: but needs a rebase already.

**Comment by pchote** [comment]
I'm not convinced about this.  On the one hand it does reduce duplication (which is a good thing), but on the other it moves a piece of gameplay-specific logic into the engine code, which is something that we want to avoid doing unless it is for performance reasons.

**Comment by atimoschenkow** [comment]
How about moving it into OpenRA.Mods.Common as an extension method for Actor?

**Comment by chrisforbes** [comment]
This would be better as extension methods provided by the gameplay code.

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/blob/master/OpenRA.Mods.RA/ActorExts.cs may be the best place at the moment.

**Comment by atimoschenkow** [comment]
Done

**Comment by Mailaender** [comment]
Squash everything together and this is good to merge from my side.

**Comment by atimoschenkow** [comment]
Done

**Comment by huwpascoe** [comment]
Good :+1:

**Comment by Mailaender** [comment]
Needs a rebase.

**Comment by atimoschenkow** [comment]
Rebased

**Comment by Mailaender** [comment]
Thanks!

**Pull request merged**